Metaclass related internal error

Hi,

We just ran into this stacktrace:

```
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.800
Traceback (most recent call last):
  File "mypy/semanal.py", line 4835, in accept
  File "mypy/nodes.py", line 950, in accept
  File "mypy/semanal.py", line 1048, in visit_class_def
  File "mypy/semanal.py", line 1125, in analyze_class
  File "mypy/semanal.py", line 1134, in analyze_class_body_common
  File "mypy/semanal.py", line 1187, in apply_class_plugin_hooks
  File "/home/travis/virtualenv/python3.8.1/lib/python3.8/site-packages/mypy_zope/plugin.py", line 266, in analyze_metaclass
    if info and any(node.fullname == expected for node in info.mro):
AttributeError: 'Var' object has no attribute 'mro'
```

I haven't had time find exactly which piece of code is causing this, but will do so as soon as I can.

EDIT: this was tested on mypy 0.800. The line which hit this was something like this:

```python
@attr.s(slots=True, frozen=True)
class Leg(metaclass=CustomMetaclass):
	a: str = attr.ib()
	b: str = attr.ib()
	c: str = attr.ib()
```

my mypy.ini:

```
[mypy]
scripts_are_modules = True
show_traceback = True
disallow_any_generics = True
warn_no_return = True
no_implicit_optional = True
ignore_missing_imports = True
namespace_packages = True
plugins = mypy_zope:plugin
```

And the run:

```shell
$ PYTHONPATH=$(pwd) mypy --show-traceback --show-error-codes --config mypy.ini z.py 
z.py:4: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.800
Traceback (most recent call last):
  File "mypy/semanal.py", line 4835, in accept
  File "mypy/nodes.py", line 950, in accept
  File "mypy/semanal.py", line 1048, in visit_class_def
  File "mypy/semanal.py", line 1125, in analyze_class
  File "mypy/semanal.py", line 1134, in analyze_class_body_common
  File "mypy/semanal.py", line 1187, in apply_class_plugin_hooks
  File "/home/arni/.config/dohop/tools_venv3/lib/python3.8/site-packages/mypy_zope/plugin.py", line 270, in analyze_metaclass
    if info and any(node.fullname == expected for node in info.mro):
AttributeError: 'Var' object has no attribute 'mro'
z.py:4: : note: use --pdb to drop into pdb
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,10 +5,10 @@ All notable changes to this project will be documented in this file.
 0.2.10 (unreleased)
 -------------------
 
-- Nothing changed yet.
+- Avoid failure when handling metaclasses of unknown type (#28).
 
 
-0.2.9 (2021-01-23)
+## [0.2.9] - 2021-01-23
 
 - Upgrade to Mypy-0.800
 - Add support for Python 3.9.

--- a/src/mypy_zope/plugin.py
+++ b/src/mypy_zope/plugin.py
@@ -260,10 +260,15 @@ class ZopeInterfacePlugin(Plugin):
     ) -> Optional[Callable[[ClassDefContext], None]]:
         # print(f"get_metaclass_hook: {fullname}")
         def analyze_metaclass(ctx: ClassDefContext) -> None:
-            metaclass = cast(NameExpr, ctx.cls.metaclass)
-            info = cast(TypeInfo, metaclass.node)
+            metaclass = ctx.cls.metaclass
+            if not isinstance(metaclass, RefExpr):
+                return
+            info = metaclass.node
+            if not isinstance(info, TypeInfo):
+                return
+
             expected = "zope.interface.interface.InterfaceClass"
-            if info and any(node.fullname == expected for node in info.mro):
+            if any(node.fullname == expected for node in info.mro):
                 self.log(f"Found zope interface: {ctx.cls.fullname}")
                 md = self._get_metadata(ctx.cls.info)
                 md["is_interface"] = True
@@ -289,7 +294,7 @@ class ZopeInterfacePlugin(Plugin):
             base_name = reason.fullname
             if not base_name:
                 return
-            if '.' not in base_name:
+            if "." not in base_name:
                 # When class is inherited from a dynamic parameter, it will not
                 # have a fully-qualified name, so we cannot use api to look it
                 # up. Just give up in this case.

--- a/tests/samples/unknown_metaclass.py
+++ b/tests/samples/unknown_metaclass.py
@@ -1,0 +1,15 @@
+"""When metaclass is not defined, ignore it"""
+
+from i.dont.exist import MetaKlass
+
+
+class Leg(metaclass=MetaKlass):
+    pass
+
+
+"""
+<output>
+unknown_metaclass.py:3: error: Cannot find implementation or library stub for module named 'i.dont.exist'
+unknown_metaclass.py:3: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
+</output>
+"""

